File: mgyoutube-multitech/api-python/modules/SearchesModule.py
from abc import ABC, abstractmethod


class SearchesModule(ABC):
    @abstractmethod
    def getSavedSearchesForParent(self,  parentUsername: str) -> list():
        pass

    @abstractmethod
    def addSearchToParent(self,  parentUsername: str, searchPhrase: str):
        pass

    @abstractmethod
    def removeSearchFromParent(self,  parentUsername: str, searchPhrase: str):
        pass

File: mgyoutube-multitech/api-go/src/modules/searchesmodule.go
package modules

// SearchesModule - the interface for a search module
type SearchesModule interface {
	GetSavedSearchesForParent(parentUsername string) ([]string, error)

	AddSearchToParent(parentUsername string, searchPhrase string) error

	RemoveSearchFromParent(parentUsername string, searchPhrase string) error
}
